scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java

```diff
@@ -0,0 +1,131 @@
+import org.ow2.proactive.scheduler.common.job.JobStatus;
+import performancetests.helper.LogProcessor;
+
+import java.io.File;
+import java.net.URL;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+
+
+
+public class TestRMReconnectioWhileRunning extends MultipleRMTBase {
+
+    static final Logger logger = Logger.getLogger(TestRMReconnectioWhileRunning.class);
+
+    private SchedulerTHelper schedulerHelper;
+
+    private static URL runningJob = TestRMReconnectioWhileRunning.class.getResource("/functionaltests/descriptors/Job_20s.xml");
+    private static URL runningJob1 = TestRMReconnectioWhileRunning.class.getResource("/functionaltests/descriptors/Job_20s-1.xml");
+    private static URL runningJob2 = TestRMReconnectioWhileRunning.class.getResource("/functionaltests/descriptors/Job_20s-2.xml");
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        if (TestScheduler.isStarted()) {
+            SchedulerTHelper.log("Killing previous scheduler.");
+            TestScheduler.kill();
+        }
+        initConfigs();
+    }
+
+
+    @Test
+    public void testTaskIsNotStuckRunning() throws Exception {
+        ProActiveConfiguration.load();
+        RMFactory.setOsJavaProperty();
+        schedulerHelper = new SchedulerTHelper(false, true);
+        schedulerHelper.createNodeSource("local", 3);
+        schedulerHelper.getSchedulerInterface().changePolicy("functionaltests.rm.PolicyWhichThrowsExceptions");
+        JobId jobId = schedulerHelper.submitJob(new File(runningJob.toURI()).getAbsolutePath());
+        JobId jobId2 = schedulerHelper.submitJob(new File(runningJob1.toURI()).getAbsolutePath());
+        JobId jobId3 = schedulerHelper.submitJob(new File(runningJob2.toURI()).getAbsolutePath());
+
+        schedulerHelper.waitForEventTaskRunning(jobId, "running_task_for20s");
+
+        Thread.sleep(200000); // sleep enough that all jobs should be able to finish
+
+        assertJobFinished(jobId2);
+        assertJobFinished(jobId3);
+        assertJobFinished(jobId);
+    }
+
+    @Test
+    public void testReconnectionIsNotHappen() throws Exception {
+        ProActiveConfiguration.load();
+        RMFactory.setOsJavaProperty();
+        schedulerHelper = new SchedulerTHelper(false, true);
+        schedulerHelper.createNodeSource("local", 3);
+        schedulerHelper.getSchedulerInterface().changePolicy("functionaltests.rm.PolicyWhichThrowsExceptions");
+        JobId jobId = schedulerHelper.submitJob(new File(runningJob.toURI()).getAbsolutePath());
+        JobId jobId2 = schedulerHelper.submitJob(new File(runningJob1.toURI()).getAbsolutePath());
+        JobId jobId3 = schedulerHelper.submitJob(new File(runningJob2.toURI()).getAbsolutePath());
+
+        schedulerHelper.waitForEventTaskRunning(jobId, "running_task_for20s");
+
+        Thread.sleep(200000); // sleep enough that all jobs should be able to finish
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)